lib/blockchain/chaindb.js

```diff
@@ -1732,6 +1818,9 @@ class ChainDB {
     if (this.options.prune)
       throw new Error('Cannot reset when pruned.');
 
+    if (this.treeState.compactionHeight !== 0)
+      throw new Error('Cannot reset when tree is compacted.');
```

Comment:
I suppose we *could* actually allow this as long as `entry.height >= compactionHeight` ?

Reply:
This is mostly meant for SPV, so decided not to disable it on tree compaction. But yeah, if node is not pruned (which is also prohibited) we could technically allow reset up to `entry.height >= compactionHeight`. Problem could occur if reset block == compactionHeight and at that point reorg is detected in that place - that could lead to issue where user can't reorg properly in the past. Ofc this is highly unlikely, but given nature of compaction I believe we should leave how it is.

Reply:
cool